easybuild/easyblocks/a/ansysem.py

```diff
@@ -0,0 +1,108 @@
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.run import run_cmd
+
+
+class EB_ANSYSEM(PackedBinary):
+    """Support for installing Ansys Electromagnetics."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize Ansys Electromagnetics specific variables."""
+        super(EB_ANSYSEM, self).__init__(*args, **kwargs)
+        self.replayfile = None
+
+    def configure_step(self):
+        """Configure Ansys Electromagnetics installation."""
+        licserv = os.getenv('EB_ANSYS_EM_LICENSE_SERVER')
+        licport = os.getenv('EB_ANSYS_EM_LICENSE_SERVER_PORT')
+        licservers = ['', '', '']
+        for i, licserver in licserv.split(','):
+            licservers[i] = licserver
+        try:
+            self.replayfile = os.path.join(self.builddir, "installer.properties")
+            txt = '\n'.join([
+                "-W Agree.selection=1",
+                "-P installLocation=\"%s\"" % self.installdir,
+                "-W TempDirectory.tempPath=\"/tmp\"",
+                "-W TempDirectory.ChangeTempPermission=\"0\"",
+                "-W LibraryOption.libraryOption=0",
+                "-W LibraryOption.libraryPath=\"\"",
+                "-W LicenseOption.licenseOption=2",
+                "-W LicenseOption.licenseFileName=\"\"",
+                "-W LicenseOption.serverCount=%s" % servercount,
+                "-W LicenseOption.serverName1=\"%s\"" % licservers[0],
+                "-W LicenseOption.serverName2=\"%s\"" % licservers[1],
+                "-W LicenseOption.serverName3=\"%s\"" % licservers[2],
+                "-W LicenseOption.tcpPort=%s" % licport,
+            ])
+            with file(self.replayfile, "w") as f:
+                f.write(txt)
+        except IOError, err:
+            raise EasyBuildError("Failed to create install properties file used for replaying installation: %s", err)
+
+    def install_step(self):
+        """Install Ansys Electromagnetics using its setup tool."""
+        cmd = "./Linux/AnsysEM/disk1/setup.exe -options \"%s\" -silent" % (self.replayfile)
+        run_cmd(cmd, log_all=True, simple=True)
+
+    def make_module_extra(self):
+        """Extra module entries for Ansys Electromagnetics."""
+        idirs = glob.glob(os.path.join(self.installdir, 'AnsysEM*/Linux*/'))
+        if len(idirs) == 1:
+            subdir = os.path.relpath(idirs[0], self.installdir)
+        else:
+            raise EasyBuildError("Failed to locate single install subdirectory AnsysEM*/Linux*/")
+
+        txt = super(EB_ANSYSEM, self).make_module_extra()
+        txt += self.module_generator.prepend_paths('PATH', subdir)
+        #txt += self.module_generator.prepend_paths('LD_LIBRARY_PATH', [os.path.join(ansysdir, 'mainwin540', 'Linux64', 'mw', 'lib-amd64_linux_optimized')])
+        #txt += self.module_generator.prepend_paths('LIBRARY_PATH', [os.path.join('ansysdir', 'mainwin540', 'Linux64', 'mw', 'lib-amd64_linux_optimized')])
```

Comment:
Not sure if these are needed, I think perhaps a user requested these in the past for compiling a custom module? It's been several years, so i'm unsure